Avoid conflict with standard string.h header on case-insensitive filesystems.
See https://github.com/torognes/vsearch/issues/91
--- src/serial/eoSerial.h.orig	2016-01-26 15:26:43.000000000 +0100
+++ src/serial/eoSerial.h	2016-01-26 15:24:02.000000000 +0100
@@ -26,7 +26,7 @@
 # include "Serializable.h"
 # include "Array.h"
 # include "Object.h"
-# include "String.h"
+# include "serial_String.h"
 # include "Parser.h"
 # include "Utils.h"
 
--- src/serial/Parser.h.orig	2016-01-26 15:26:49.000000000 +0100
+++ src/serial/Parser.h		2016-01-26 15:24:11.000000000 +0100
@@ -23,7 +23,7 @@
 # define __EOSERIAL_PARSER_H__
 
 # include "Entity.h"
-# include "String.h"
+# include "serial_String.h"
 # include "Object.h"
 
 /**
--- src/serial/Parser.cpp.orig	2016-01-26 15:26:50.000000000 +0100
+++ src/serial/Parser.cpp	2016-01-26 15:24:17.000000000 +0100
@@ -25,7 +25,7 @@
 
 # include "Array.h"
 # include "Object.h"
-# include "String.h"
+# include "serial_String.h"
 
 // in debug mode only
 // # define DEBUG(x) std::cout << x << std::endl;
--- src/serial/String.cpp.orig	2016-01-26 15:26:57.000000000 +0100
+++ src/serial/String.cpp	2016-01-26 15:24:28.000000000 +0100
@@ -19,7 +19,7 @@
 Authors:
     Benjamin Bouvier <benjamin.bouvier@gmail.com>
 */
-# include "String.h"
+# include "serial_String.h"
 
 namespace eoserial
 {
--- src/serial/Utils.h.orig	2016-01-26 15:26:59.000000000 +0100
+++ src/serial/Utils.h		2016-01-26 15:24:37.000000000 +0100
@@ -24,7 +24,7 @@
 
 # include "Array.h"
 # include "Object.h"
-# include "String.h"
+# include "serial_String.h"
 
 namespace eoserial
 {
